app/views/common/os_selection/_pxe_loader.html.erb

```diff
@@ -0,0 +1,12 @@
+<%= fields_for item do |f| -%>
+  <% loaders = @operatingsystem.available_loaders(item.architecture); if loaders.any? -%>
+    <% selected_attr = f.object.new_record? && item.hostgroup.present? && item.hostgroup.pxe_loader.nil? ? { :selected => @operatingsystem.preferred_loader(item.architecture) } : {} -%>
```

Comment:
> I will take a look tomorrow.

Sorry for being a pain, but it should now be possible to set the sane default in the model. I think, that's way cleaner than setting it in the controller / view.